README.md

```diff
@@ -27,6 +27,36 @@ The `build` script takes arguments which may be used to customize the build.
 - `--source-only`: Only build source archive
 - `--binary-only`: Only build the binary archives
 
+## Package build API
+
+Package builds make use of shell functions which are defined in this repo.
+
+### `prepare_source`
+
+`apt_src [--ignore-orig] $package_name`
+
+Use source code from a given apt source package.
+
+`git_src $git_url`
+
+Use source code from a remote git repository
+
+`import_src $path`
+
+Use source code from a local path (todo: how does that work?)
+
+`apply_patched [patch_dir(default=./patches)]`
+
+Apply patches from a directory.
+By convention the directory is called `patches`.
+Another name may be chosen and passed as a parameter.
+
+The `patches` directory requires a `series` file listing the patches in the order to be applied, and the patch files.
+
+`import_upstream_patches [patch_dir(default=./upstream_patches)]`
+
+### `prepare_binary`
```

Comment:
Content for prepare binary missing.

The package pipeline has three stages.
1. source (uses [bin/source](https://github.com/gardenlinux/package-build/blob/main/bin/source))
2. binary (uses [bin/binary](https://github.com/gardenlinux/package-build/blob/main/bin/binary))
3. release (defined in GH actions only)

For package builds that need special treatment, we have introduced hooks where we can add additional steps
- [prepare_source](https://github.com/gardenlinux/package-build/blob/70a02624911842d53a059ef96c8bbc9167c76de3/bin/source#L103) hook is executed in stage 1
- [prepare_binary](https://github.com/gardenlinux/package-build/blob/70a02624911842d53a059ef96c8bbc9167c76de3/bin/binary#L25) hook is executed in stage 2